server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchAction.java

```diff
@@ -124,43 +123,38 @@ private SearchRequest createRequest() {
       return request;
     }
 
-    private List<MetricDto> searchMetrics() {
-      List<MetricDto> dbMetrics = dbClient.metricDao().selectByKeys(dbSession, request.getMetricKeys());
-      List<String> metricKeys = dbMetrics.stream().map(MetricDto::getKey).collect(toList());
-      checkRequest(request.getMetricKeys().size() == dbMetrics.size(), "The following metrics are not found: %s",
-        String.join(", ", difference(request.getMetricKeys(), metricKeys)));
-      return dbMetrics;
-    }
-
     private List<ComponentDto> searchProjects() {
       return getAuthorizedProjects(searchByProjectKeys(dbSession, request.getProjectKeys()));
     }
 
-    private List<ComponentDto> getAuthorizedProjects(List<ComponentDto> projectDtos) {
+    private List<ComponentDto> searchByProjectKeys(DbSession dbSession, List<String> projectKeys) {
+      return dbClient.componentDao().selectByKeys(dbSession, projectKeys);
+    }
+
+    private List<ComponentDto> getAuthorizedProjects(List<ComponentDto> componentDtos) {
       if (userSession.isRoot()) {
         // the method AuthorizationDao#keepAuthorizedProjectIds() should be replaced by
         // a call to UserSession, which would transparently support roots.
         // Meanwhile root is explicitly handled.
-        return projectDtos;
+        return componentDtos;
       }
+      Set<String> projectUuids = componentDtos.stream().map(ComponentDto::projectUuid).collect(Collectors.toSet());
+      List<ComponentDto> projectDtos = dbClient.componentDao().selectByUuids(dbSession, projectUuids);
       Map<String, Long> projectIdsByUuids = projectDtos.stream().collect(uniqueIndex(ComponentDto::uuid, ComponentDto::getId));
       Set<Long> authorizedProjectIds = dbClient.authorizationDao().keepAuthorizedProjectIds(dbSession,
```

Comment:
it's this line that prevent to load measures for files, directory and only for component of Scope PROJECT?